### DISCUSSION TITLE
should update pediacluster status when conditions or SyncResources changed rather than timed execution

### DISCUSSION THREAD

**Discussion by duanmengkk** [clusterpedia-io/clusterpedia#341] [Ideas]
![image](https://user-images.githubusercontent.com/17697155/188404916-4aadded7-2bd6-4188-a64b-43cce7cea976.png)
These logs make me confused.At first I thought the cluster status was constantly changing(sync resources Or cluster healthy status),and I research the code and find the status of pediacluster is updated every 5s.See code [https://github.com/clusterpedia-io/clusterpedia/blob/b06cc6d0f6f06af50269a9533202211552c63252/pkg/synchromanager/clustersynchro/cluster_monitor.go#L36](url)

Do you need to optimize it? @Iceber

**Comment by Iceber** [comment]
What is the version of clusterpedia? Although it is checked once every 5 seconds, the status update log is not printed if the status has not changed.

Can you give me a more complete log, the picture has cut off part of the log.

**Reply by duanmengkk** [reply]
![image](https://user-images.githubusercontent.com/17697155/188415475-a4b7161e-e643-48de-8889-af798492db9d.png)
The latest code in main branch.

**Reply by duanmengkk** [reply]
![image](https://user-images.githubusercontent.com/17697155/188419979-88f2b296-af8a-4fd2-a0d2-0875e4be2c37.png)
The `lastTransitionTime` between latest status and updated status is diffrent

**Comment by Iceber** [comment]
I'm running clustersynchro manager with the main branch
```bash



I0905 18:01:22.612425    8227 clustersynchro_manager.go:498] "Update Cluster Status" cluster="cluster-example" conditions=[{Type:Validated Status:True ObservedGeneration:0 LastTransitionTime:2022-08-02 12:46:34 +0800 CST Reason:Validated Message:} {Type:SynchroRunning Status:True ObservedGeneration:0 LastTransitionTime:2022-09-05 18:01:14 +0800 CST Reason:Running Message:cluster synchro is running} {Type:ClusterHealthy Status:True ObservedGeneration:0 LastTransitionTime:2022-09-05 18:01:15 +0800 CST Reason:Healthy Message:cluster health responded with ok} {Type:Ready Status:True ObservedGeneration:0 LastTransitionTime:2022-09-05 18:01:15 +0800 CST Reason:Ready Message:}]



Now: 09-05 18:03:18
```
I did not find frequent status updates.

Can you print the version information?
```bash
# replace the pod name of clusterpedia clustersynchro-manager
$ kubectl -n clusterpedia-system exec {clustersynchro-manager-pod-name} -- clustersynchro-manager --version=raw
```
In a previous commit, it did trigger frequent updates, but this has been fixed.

**Reply by duanmengkk** [reply]
I run it in my local env. I guess the status in [ResourceSynchro](https://github.com/clusterpedia-io/clusterpedia/blob/b06cc6d0f6f06af50269a9533202211552c63252/pkg/synchromanager/clustersynchro/resource_synchro.go#L45)  and status in [pediaclusterListener ](https://github.com/clusterpedia-io/clusterpedia/blob/b06cc6d0f6f06af50269a9533202211552c63252/pkg/synchromanager/clustersynchro_manager.go#L452) may have some time delay. The informer doesn't handle event in time,so the status get from pediaclusterListener is always delayed by few seconds.

It is just my  guess,not sure.
![image](https://user-images.githubusercontent.com/17697155/188435547-31df5b72-ecdc-4ab9-aca3-580dea1fc00d.png)
![image](https://user-images.githubusercontent.com/17697155/188435616-7ac07deb-618b-4163-aa5e-75c3872d7500.png)

**Reply by Iceber** [reply]
The infromer delay may result in one or two additional status updates, but not a continuous status update

Do other environments have this problem?
